src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1945,16 +1923,29 @@ export default class AircraftModel {
      * @return {number} speed, in knots
      */
     _calculateTargetedSpeedVnav() {
-        const hardRestrictedWaypointModel = this.fms.nextHardSpeedRestrictedWaypoint;
+        const nextSpeedMaximumWaypoint = this.fms.getNextWaypointWithMaximumSpeedAtOrBelow(this.speed);
+        const nextSpeedMinimumWaypoint = this.fms.getNextWaypointWithMinimumSpeedAtOrAbove(this.speed);
+        const maximumSpeedExists = !_isNil(nextSpeedMaximumWaypoint);
+        const minimumSpeedExists = !_isNil(nextSpeedMinimumWaypoint);
```

Comment:
these could be re-titled to `hasMaxSpeed` and `hasMinSpeed`, but that's a nit-pick.  just something to call out.

Reply:
eh. why not, right!?!?

done :smile: